### PR TITLE
code-cursor: 0.43.0 -> 0.43.5

### PR DESCRIPTION
**Pull request by fazo96**
This is a bug fix only update, and the cursor team does not release changelogs for them. Main changelog is here: https://changelog.cursor.com/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by sarahec** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 360159`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>code-cursor</li>
  </ul>
</details>

**Review by sarahec — APPROVED** [review]

File: pkgs/by-name/co/code-cursor/package.nix
Comment (LunNova) [inline pkgs/by-name/co/code-cursor/package.nix:14]: Could mirror to web.archive.org and add that URL too so if this dies the derivation still works, since it seems they don't stick around forever.

Comment (sarahec) [reply pkgs/by-name/co/code-cursor/package.nix:14]: The cursor team is setting up an archive of old builds for people to roll back to. I and several others made enough noise that they finally chose to do the right thing.

Comment (sarahec) [reply pkgs/by-name/co/code-cursor/package.nix:14]: @LunNova unfortunately, your comment is blocking merging. Can you approve, please?

**Comment by sarahec** [comment]
Requested a committer on Matrix so we can close this out.

**Comment by sarahec** [comment]
Replaced by #362556

**Comment by sarahec** [comment]
Two quick follow-ups:

@LunNova You had a great idea and thankfully we will have a proper place to download updates (and rollbacks) soon. Unfortunately, the system treats open "suggestions" as mandatory -- all other approvals are ignored and this PR languished. In the future, an ordinary comment won't block a PR. (Please don't be discouraged.)

@fazo96 While I was trying to get a committer to merge this, the automatic update bot came through with a later version. It takes basically no time to review and get the update merged which is why I closed this PR. (It's no reflection on your work, just that we had all had a quicker path to getting `cursor` updated.) 

Please don't hesitate to create and/or review PRs in the future. We all have a learning curve and you'll find yourself becoming an expert sooner than you think.

File: pkgs/by-name/co/code-cursor/package.nix
Comment (LunNova) [reply pkgs/by-name/co/code-cursor/package.nix:14]: Wait why was my ordinary comment as review blocking approval? It wasn't "needs changes" so shouldn't impact that. :confused:

Comment (sarahec) [reply pkgs/by-name/co/code-cursor/package.nix:14]: It was entered as a recommended change (delete the URL plus some text). The person who keeps the "ready to merge" tags up to date sees this as "it's not ready yet"). 

Another way to solve this would have been to mark this comment as resolved.

**Comment by fazo96** [comment]
No worries at all, makes sense to close this. Thanks!

**Pull request closed without merging**
